dsc/tests/dsc_variables.tests.ps1

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+Describe 'Configruation variables tests' {
+    It 'Variables example config works' {
+        $configFile = "$PSSCriptRoot/../examples/variables.dsc.yaml"
+        $out = dsc config get -p $configFile | ConvertFrom-Json
+        $LASTEXITCODE | Should -Be 0
+        $out.results[0].result.actualState.output | Should -BeExactly 'myOutput is: Hello world!, myObject is: baz'
+    }
+
+    It 'Duplicated variable takes last value' {
+        $configYaml = @'
+$schema: https://raw.githubusercontent.com/PowerShell/DSC/main/schemas/2024/04/config/document.json
+variables:
+  myVariable: foo
+  myVariable: bar
+resources:
+- name: test
+  type: Test/Echo
+  properties:
+    output: "[variables('myVariable')]"
+'@
+        $out = dsc config get -d $configYaml | ConvertFrom-Json
+        Write-Verbose -Verbose $out
+        $LASTEXITCODE | Should -Be 0
+        $out.results[0].result.actualState.output | Should -Be 'bar'
+    }
+
+    It 'Missing variable returns error' {
```

Comment:
The essence of a `variable` in any language is that it's value can be changed; I don't see such example or tests.

Reply:
You are correct, but that's not how it's used in an ARM template